depthai_examples/ros1_src/stereo_inertial_publisher.cpp

```diff
@@ -374,32 +376,57 @@ int main(int argc, char** argv) {
                                                         nnPath);
 
     std::shared_ptr<dai::Device> device;
-    std::vector<dai::DeviceInfo> availableDevices = dai::Device::getAllAvailableDevices();
-
-    std::cout << "Listing available devices..." << std::endl;
-    for(auto deviceInfo : availableDevices) {
-        std::cout << "Device Mx ID: " << deviceInfo.getMxId() << std::endl;
-        if(deviceInfo.getMxId() == mxId) {
-            if(deviceInfo.state == X_LINK_UNBOOTED || deviceInfo.state == X_LINK_BOOTLOADER) {
-                isDeviceFound = true;
-                if(poeMode) {
-                    device = std::make_shared<dai::Device>(pipeline, deviceInfo);
-                } else {
-                    device = std::make_shared<dai::Device>(pipeline, deviceInfo, usb2Mode);
-                }
-                break;
-            } else if(deviceInfo.state == X_LINK_BOOTED) {
-                throw std::runtime_error("ros::NodeHandle() from Node \"" + pnh.getNamespace() + "\" DepthAI Device with MxId  \"" + mxId
-                                         + "\" is already booted on different process.  \"");
+    if(useWithIP) // Connecting to a camera with specific IP, requires static IP config beforehand
+    {
+        auto deviceInfo = dai::DeviceInfo(ipAddress);
+        if(deviceInfo.state == X_LINK_UNBOOTED || deviceInfo.state == X_LINK_BOOTLOADER || deviceInfo.state == X_LINK_FLASH_BOOTED) {
```

Comment:
The value of `deviceInfo.state` is always `0` here so the exception "Could NOT connect to device with IP Address" is always thrown. If we want to check if another process already has a connection, maybe we should do something like this:
```c++
    if(useWithIP) {
        auto connectionInfo = dai::Device::getAllConnectedDevices();
        for (auto const& info : connectionInfo)
        {
            if (info.name == ipAddress && info.state == X_LINK_BOOTED) {
                // Throw exception here....
            }
        }
        auto deviceInfo = dai::DeviceInfo(ipAddress);
        if(poeMode) {
            device = std::make_shared<dai::Device>(pipeline, deviceInfo);
        } else {
            device = std::make_shared<dai::Device>(pipeline, deviceInfo, usb2Mode);
        }
        if(device.getDeviceInfo().state == X_LINK_BOOTED) {
            std::cout << "Device found with IP Address: " << ipAddress <<  std::endl;
        }
    else if(useWithMxId) {
        // ...
    }
```

Reply:
@zflat Did you assign a static ip to your camera using depthai device manager? I tested it with my camera and it seems to be working. deviceInfo.state is actually an enum, so if it returns 1, it can take the following values
![image](https://user-images.githubusercontent.com/30050658/201364721-c9f2af51-4f9e-4676-9bc9-c6e3041d9775.png)
so 1 probably corresponds to ```X_LINK_BOOTED```, which kinda tells me that you have not used device manager, because once you use device manager to assign static IP, ```deviceInfo.state``` becomes X_LINK_FLASH_BOOTED.

Reply:
Yes, I did assign a static IP.
I tested this code on an OAK-1 (actually copy and pasted the logic to `rgb_publisher.cpp` because it can't use a pipeline that is set up for steroe) and also an OAK-D.

OAK-1
![Screenshot from 2022-11-09 16-29-06](https://user-images.githubusercontent.com/1305536/201390527-3fe29355-2d66-4ea8-9846-48ed6a67a49d.png)


OAK-D
![Screenshot from 2022-11-11 12-08-53](https://user-images.githubusercontent.com/1305536/201393385-33d9544c-df38-4237-8687-e2739e98d484.png)

Reply:
@zflat  Could you try to flash the newest bootloader? I remember I had a problem connecting to my device with IP address before updating the bootloader. If it doesn't solve the issue, I should have OAK (Series 1) POE lying somewhere. I will test it with that guy.

Reply:
I have tried flashing a new bootloader and I just get an error popup and then a message to the terminal that it was cancelled. Sorry I can't be of more help.

Reply:
@zflat  No problem! Lemme give it a try.

Reply:
@zflat  You are right. Somehow, ```deviceInfo.state``` returns 0 once it is not created by get all available devices. 0 means X_LINK_ANY_STATE, so I added that one to if condition and tested it with two cameras at the same time. It should be working fine now.

Reply:
@zflat  what was your execution command here ?